File: clojure/06-io-operations/input.clj
;; Reading console input in Clojure

;; read-line reads one line from stdin as a string.
;; with-in-str feeds it fixed input so the example is reproducible.
(with-in-str "Grace\n42\n"
  (let [name (read-line)
        age  (read-line)]
    (println (str "Hello, " name "!"))
    (println (str "Next year you will be "
                  (inc (Integer/parseInt age))
                  "."))))

;; In a real interactive program you would prompt and read directly:
;;   (print "Enter your name: ")
;;   (flush)                       ; force the prompt to appear before reading
;;   (let [name (read-line)]
;;     (println (str "Hello, " name "!")))

File: ocaml/06-io-operations/input.ml
let () =
  print_string "Enter your name: ";
  let name = read_line () in

  print_string "Enter your age: ";
  let age = read_int () in

  Printf.printf "Hello, %s!\n" name;
  Printf.printf "Next year you will be %d\n" (age + 1)
